cmd/prometheus/main.go

```diff
@@ -294,8 +296,8 @@ func main() {
 	}
 
 	// Unregister the default GoCollector, and reregister with our defaults.
-	if prometheus.Unregister(collectors.NewGoCollector()) {
-		prometheus.MustRegister(
+	if registry.Unregister(collectors.NewGoCollector()) {
```

Comment:
The good thing with the custom registry is that it doesn't even have a Go collector pre-registered, so you can remove the `if` altogether and register the custom Go collector unconditionally.